pkg/webhookconfig/configmanager.go

```diff
@@ -234,15 +231,44 @@ func (m *webhookConfigManager) deletePolicy(obj interface{}) {
 	m.enqueue(p)
 }
 
-func (m *webhookConfigManager) deleteWebhook(obj interface{}) {
-	m.log.WithName("deleteWebhook").Info("resource webhook configuration was deleted, recreating...")
-	if webhook, ok := obj.(*unstructured.Unstructured); ok {
-		k := webhook.GetKind()
-		if (k == kindMutating && webhook.GetName() == config.MutatingWebhookConfigurationName) ||
-			(k == kindValidating && webhook.GetName() == config.ValidatingWebhookConfigurationName) {
-			m.enqueueAllPolicies()
+func (m *webhookConfigManager) deleteMutatingWebhook(obj interface{}) {
+	m.log.WithName("deleteMutatingWebhook").Info("resource webhook configuration was deleted, recreating...")
+	webhook, ok := obj.(*admregapi.MutatingWebhookConfiguration)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			m.log.Info("Couldn't get object from tombstone", "obj", obj)
+			return
+		}
+		webhook, ok = tombstone.Obj.(*admregapi.MutatingWebhookConfiguration)
+		if !ok {
+			m.log.Info("tombstone contained object that is not a MutatingWebhookConfiguration", "obj", obj)
+			return
 		}
 	}
+	if webhook.GetName() == config.MutatingWebhookConfigurationName {
+		m.enqueueAllPolicies()
```

Comment:
Looks like we don't need to queue all policies, one should be enough :thinking:

Reply:
cc @realshuting

Reply:
Would it be better to enqueue a single policy on its events? Since this PR switched to typed client, there would be no update failure anymore.

Reply:
I think I will rework the controller to simply have a reconcile function.
No need to requeue anything IMHO.